node/src/keyshare/mod.rs

```diff
@@ -58,22 +52,17 @@ pub enum KeyshareStorageFactory {
 }
 
 impl KeyshareStorageFactory {
-    pub async fn create(&self) -> anyhow::Result<Box<dyn KeyshareStorage>> {
+    pub async fn create(&self) -> anyhow::Result<impl KeyshareStorage> {
```

Comment:
How is this going to be implementable if Gcp is also going to be supported (and as Bowen said we need to support it)? impl KeyshareStorage is a concrete type; there's no way you can return two different KeyshareStorage types.

I think one idea is to have KeyshareStorage still be a dyn trait (use dyn everywhere), but have load and store take an ID - they will return Vec<u8>. Then, have another wrapper on top which takes a keyshare type parameter and returns the deserialized type of the keyshare. That way we can abstract away the storage of the keyshare with a dyn trait, but still have type safety.